dev/io.openliberty.jaxws.config_fat/test-applications/simpleTestService/src/io/openliberty/jaxws/fat/stubclient/client/SimpleEchoService.java

```diff
@@ -11,29 +11,41 @@
 
 import java.net.MalformedURLException;
 import java.net.URL;
+import java.util.logging.Logger;
+
 import javax.xml.namespace.QName;
 import javax.xml.ws.Service;
 import javax.xml.ws.WebEndpoint;
 import javax.xml.ws.WebServiceClient;
 import javax.xml.ws.WebServiceException;
 import javax.xml.ws.WebServiceFeature;
 
-@WebServiceClient(name = "SimpleEchoService", targetNamespace = "http://stubclient.fat.jaxws.openliberty.io/", wsdlLocation = "WEB-INF/wsdl/SimpleEcho.wsdl")
+@WebServiceClient(name = "SimpleEchoService", targetNamespace = "http://stubclient.fat.jaxws.openliberty.io/")
 public class SimpleEchoService
     extends Service
 {
 
     private static final URL SIMPLEECHOSERVICE_WSDL_LOCATION;
     private static final WebServiceException SIMPLEECHOSERVICE_EXCEPTION;
     private static final QName SIMPLEECHOSERVICE_QNAME = new QName("http://stubclient.fat.jaxws.openliberty.io/", "SimpleEchoService");
-
+    private final static Logger logger = Logger.getLogger(SimpleEchoService.class.getName());
+    
     static {
         URL url = null;
         WebServiceException e = null;
+        String urlString = null;
         try {
             URL baseUrl = SimpleEchoService.class.getClassLoader().getResource(".");
-            url = new URL(baseUrl, "../wsdl/SimpleEcho.wsdl");
+            String host = System.getProperty("hostName");
+            if (host == null) {
+                logger.info("Failed to obtain host from system property, hostName, falling back to localhost");
+                host = "localhost";
+            }
+            urlString = new StringBuilder().append("http://" + host + ":").append(Integer.getInteger("bvt.prop.HTTP_default")).append("/simpleTestService/SimpleEchoService?wsdl").toString();
+            logger.info("URL : " + urlString + " will be used for SimpleEchoService.");
+            url = new URL(baseUrl, urlString);
```

Comment:
Is this URL right, seems like it will take the classloader path plus the url string and combine them?

Reply:
You can should just call `URL(urlString)`

Reply:
It worked both on my local and PB. But I can change it.